File: app/src/androidTest/java/art/manguste/android/ArasakaWeapons/MainActivityRecyclerViewTest.java
package art.manguste.android.ArasakaWeapons;

import androidx.test.espresso.contrib.RecyclerViewActions;
import androidx.test.espresso.matcher.ViewMatchers;
import androidx.test.rule.ActivityTestRule;
import org.junit.Rule;
import org.junit.Test;
import static androidx.test.espresso.Espresso.onView;
import static androidx.test.espresso.action.ViewActions.click;
import static androidx.test.espresso.assertion.ViewAssertions.matches;
import static androidx.test.espresso.matcher.ViewMatchers.isDisplayed;
import static androidx.test.espresso.matcher.ViewMatchers.withId;
import static androidx.test.espresso.matcher.ViewMatchers.withText;
import static org.hamcrest.core.AllOf.allOf;

public class MainActivityRecyclerViewTest {

    @Rule
    public ActivityTestRule<MainActivity> mActivityTestRule = new ActivityTestRule<>(MainActivity.class);

    @Test
    public void clickRecyclerViewItem_OpensDetailActivity() {
        String weaponName = mActivityTestRule.getActivity().getString(R.string.weapon_gorilla_title);

        // click on displayed recycler view, on a first item
        onView(allOf(ViewMatchers.withId(R.id.recycler_view), isDisplayed()))
                .perform(RecyclerViewActions.actionOnItemAtPosition(0, click()));

        onView(withId(R.id.productName)).check(matches(withText(weaponName)));
    }
}

File: app/src/androidTest/java/art/manguste/android/arasakaWeapons/MainActivityRecyclerViewTest.kt
package art.manguste.android.arasakaWeapons

import androidx.recyclerview.widget.RecyclerView
import androidx.test.espresso.Espresso
import androidx.test.espresso.action.ViewActions
import androidx.test.espresso.assertion.ViewAssertions
import androidx.test.espresso.contrib.RecyclerViewActions
import androidx.test.espresso.matcher.ViewMatchers
import androidx.test.rule.ActivityTestRule
import org.hamcrest.core.AllOf
import org.junit.Rule
import org.junit.Test

class MainActivityRecyclerViewTest {
    @Rule
    var mActivityTestRule = ActivityTestRule(MainActivity::class.java)
    @Test
    fun clickRecyclerViewItem_OpensDetailActivity() {
        val weaponName = mActivityTestRule.activity.getString(R.string.weapon_gorilla_title)

        // click on displayed recycler view, on a first item
        Espresso.onView(AllOf.allOf(ViewMatchers.withId(R.id.recyclerView), ViewMatchers.isDisplayed()))
                .perform(RecyclerViewActions.actionOnItemAtPosition<RecyclerView.ViewHolder>(0, ViewActions.click()))
        Espresso.onView(ViewMatchers.withId(R.id.productName)).check(ViewAssertions.matches(ViewMatchers.withText(weaponName)))
    }
}